src/python/pants/core/util_rules/environments.py

```diff
@@ -363,5 +319,130 @@ def extract_process_config_from_environment(tgt: EnvironmentTarget) -> ProcessCo
     return ProcessConfigFromEnvironment(docker_image=docker_image)
 
 
+logger = logging.getLogger(__name__)
+
+
+class EnvironmentSensitiveOptionFieldMixin(ABC):
```

Comment:
Not sure `ABC` is necessary here? It makes the inheritance chain harder to follow